### PR TITLE
Add support for Cypress Modus Toolbox

### PR DESCRIPTION
**Pull request by winkj**
I've tried this with SGP and SHT. 

This should definitely see some device testing before being merged

### CODE REVIEW

**Comment by rnestler** [comment]
What is the difference between this PR and #17? Just another Cypress library that is used?

**Comment by winkj** [comment]
Apologies for the delay. This is a different I2C stack; both are from Cypress

File: hw_i2c/sample-implementations/cypress_modus_toolbox/sensirion_hw_i2c_implementation.c [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+		do
+		{
+			masterStatus  = Cy_SCB_I2C_MasterGetStatus(mI2C_HW, &mI2C_context);
+			Cy_SysLib_DelayUs(CY_SCB_WAIT_1_UNIT);
+			timeout--;
+
+		} while ((0UL != (masterStatus & CY_SCB_I2C_MASTER_BUSY)) && (timeout > 0));
+
+	   if (timeout <= 0)
+		{
```
Comment (rnestler) [inline hw_i2c/sample-implementations/cypress_modus_toolbox/sensirion_hw_i2c_implementation.c]: The whitespace looks a bit odd in the file. We recently added clang-format support to the repositories, you may use that to format the files. See https://github.com/Sensirion/embedded-sht/blob/master/.clang-format

File: hw_i2c/sample-implementations/cypress_modus_toolbox/sensirion_hw_i2c_implementation.c [hunk 2]
```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright (c) 2017, Sensirion AG
```
Comment (rnestler) [inline hw_i2c/sample-implementations/cypress_modus_toolbox/sensirion_hw_i2c_implementation.c]: Maybe use 2019 as the copyright year.

File: hw_i2c/sample-implementations/cypress_modus_toolbox/sensirion_hw_i2c_implementation.c [hunk 3]
```diff
@@ -0,0 +1,231 @@
...
+	cy_stc_sysint_t mI2C_SCB_IRQ_cfg =
+	{
+			/*.intrSrc =*/ scb_3_interrupt_IRQn,
+			/*.intrPriority =*/ 2UL
+	};
+	/* Initialize and enable the I2C in master mode. */
+	initStatus = Cy_SCB_I2C_Init(mI2C_HW, &mI2C_config, &mI2C_context);
+	if(initStatus != CY_SCB_I2C_SUCCESS)
+	{
+		// return STATUS_FAIL;
```
Comment (rnestler) [inline hw_i2c/sample-implementations/cypress_modus_toolbox/sensirion_hw_i2c_implementation.c]: @abrauchli We should really change the interface to add error codes for the `sensirion_i2c_init` function.

File: hw_i2c/sample-implementations/cypress_modus_toolbox/sensirion_hw_i2c_implementation.c [hunk 4]
```diff
@@ -0,0 +1,231 @@
...
+
+	/* Initiate write transaction. */
+	errorStatus = Cy_SCB_I2C_MasterWrite(mI2C_HW, &transferConfig, &mI2C_context);
+	if(errorStatus == CY_SCB_I2C_SUCCESS)
+	{
+		/* Wait until master complete read transfer or time out has occurred */
+		do
+		{
+			masterStatus  = Cy_SCB_I2C_MasterGetStatus(mI2C_HW, &mI2C_context);
+			Cy_SysLib_DelayUs(CY_SCB_WAIT_1_UNIT);
```
Comment (rnestler) [inline hw_i2c/sample-implementations/cypress_modus_toolbox/sensirion_hw_i2c_implementation.c]: The `CY_SCB_WAIT_1_UNIT` is really just 1us? Did you check that the timeout kicks in after ~1s?

File: hw_i2c/sample-implementations/cypress_modus_toolbox/sensirion_hw_i2c_implementation.c (hunk 3 above)
Comment (abrauchli) [reply hw_i2c/sample-implementations/cypress_modus_toolbox/sensirion_hw_i2c_implementation.c]: yes, let's

**Comment by rnestler** [comment]
@winkj Is this still active? Otherwise I'll close it.

**Comment by winkj** [comment]
Closing for the time being, let's revisit after we discuss overall driver strategy

**Pull request closed without merging**